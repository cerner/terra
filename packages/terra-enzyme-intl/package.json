{
  "name": "@cerner/terra-enzyme-intl",
  "version": "4.6.1",
  "description": "Enzyme helpers for fortifying tests that depend on react-intl by decoupling the need for actual translations.",
  "author": "Cerner Corporation",
  "main": "lib/index.js",
  "engines": {
    "node": ">=10"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/cerner/terra.git",
    "directory": "packages/terra-enzyme-intl"
  },
  "bugs": {
    "url": "https://github.com/cerner/terra/issues"
  },
  "homepage": "https://github.com/cerner/terra/packages/terra-enzyme-intl#readme",
  "publishConfig": {
    "access": "public"
  },
  "license": "Apache-2.0",
  "keywords": [
    "Cerner",
    "enzyme",
    "i18n",
    "intl",
    "jest",
    "react",
    "react-intl",
    "Terra",
    "terra-enzyme-intl"
  ],
  "files": [
    "lib"
  ],
  "dependencies": {
    "prop-types": "^15.5.8"
  },
  "devDependencies": {
    "react-intl": "^5.25.1"
  },
  "peerDependencies": {
    "enzyme": "3.x",
    "react": "16.x",
    "react-intl": "^5.25.1"
  },
  "scripts": {
    "clean:install": "npm run clean && npm install",
    "compile": "babel --root-mode upward src --out-dir lib --copy-files",
    "jest": "jest --config ../../jest.config.js",
    "lint": "eslint --ext .js,.jsx . --ignore-path ../../.eslintignore",
    "precompile": "rm -rf lib",
    "test": "npm run lint && npm run jest"
  }
}
