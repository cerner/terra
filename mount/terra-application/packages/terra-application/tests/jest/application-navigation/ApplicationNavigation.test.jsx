import React from 'react';
import { shallowWithIntl } from '@cerner/terra-enzyme-intl';
import ApplicationNavigation from '../../../src/application-navigation/ApplicationNavigation';

jest.mock('react-intl', () => {
  const reactIntl = jest.requireActual('react-intl');
  const intl = reactIntl.createIntl({
    locale: 'en',
  });

  return {
    ...reactIntl,
    useIntl: () => intl,
  };
});

describe('ApplicationNavigation', () => {
  it('should render default element', () => {
    const shallowComponent = shallowWithIntl(
      <ApplicationNavigation
        id="test-application-id"
      />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render with all props', () => {
    const shallowComponent = shallowWithIntl(
      <ApplicationNavigation
        id="test-application-id"
        activeBreakpoint="large"
        titleConfig={{
          title: 'test-title',
        }}
        activeNavigationItemKey="my-test-key"
        hero={<div>my test hero</div>}
        userConfig={{
          name: 'user-name',
          detail: 'user-detail',
          initials: 'user-initials',
          imageSrc: 'user-src',
        }}
        userActionConfig={{
          text: 'Edit Photo',
          userActionCallback: () => jest.fn(),
        }}
        navigationItems={[{ text: 'test-text', key: 'my-test-key' }]}
        extensionItems={[{ text: 'test-text-1', key: 'my-test-key-1', icon: <span>my icon</span> }]}
        utilityItems={[{ text: 'test-text-2', key: 'my-test-key-2', icon: <span>my icon</span> }]}
        notifications={{ 'my-test-key-1': 3 }}
        navigationRenderFunction={jest.fn()}
        onSelectNavigationItem={jest.fn()}
        onSelectExtensionItem={jest.fn()}
        onSelectUtilityItem={jest.fn()}
        onSelectSettings={jest.fn()}
        onSelectHelp={jest.fn()}
        onSelectLogout={jest.fn()}
        onDrawerMenuStateChange={jest.fn()}
      >
        <div>test child content</div>
      </ApplicationNavigation>,
    );

    expect(shallowComponent).toMatchSnapshot();
  });
});
