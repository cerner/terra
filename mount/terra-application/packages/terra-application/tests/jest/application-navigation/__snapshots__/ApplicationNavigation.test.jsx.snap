// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationNavigation should render default element 1`] = `
<ApplicationNavigation
  id="test-application-id"
  notifications={Object {}}
  utilityItems={Array []}
>
  <ApplicationLoadingOverlayProvider>
    <ApplicationStatusOverlayProvider>
      <withPromptRegistration(NavigationPromptCheckpoint)>
        <injectIntl(ApplicationErrorBoundary)>
          <Suspense
            fallback={
              <ApplicationLoadingOverlay
                backgroundStyle="clear"
                isOpen={true}
              />
            }
          />
        </injectIntl(ApplicationErrorBoundary)>
      </withPromptRegistration(NavigationPromptCheckpoint)>
    </ApplicationStatusOverlayProvider>
  </ApplicationLoadingOverlayProvider>
</ApplicationNavigation>
`;

exports[`ApplicationNavigation should render with all props 1`] = `
<ApplicationNavigation
  activeNavigationItemKey="my-test-key"
  extensionItems={
    Array [
      Object {
        "icon": <span>
          my icon
        </span>,
        "key": "my-test-key-1",
        "text": "test-text-1",
      },
    ]
  }
  hero={
    <div>
      my test hero
    </div>
  }
  id="test-application-id"
  navigationItems={
    Array [
      Object {
        "key": "my-test-key",
        "text": "test-text",
      },
    ]
  }
  notifications={
    Object {
      "my-test-key-1": 3,
    }
  }
  onDrawerMenuStateChange={[MockFunction]}
  onSelectExtensionItem={[MockFunction]}
  onSelectHelp={[MockFunction]}
  onSelectLogout={[Function]}
  onSelectNavigationItem={[Function]}
  onSelectSettings={[MockFunction]}
  onSelectUtilityItem={[MockFunction]}
  titleConfig={
    Object {
      "title": "test-title",
    }
  }
  userActionConfig={
    Object {
      "text": "Edit Photo",
      "userActionCallback": [Function],
    }
  }
  userConfig={
    Object {
      "detail": "user-detail",
      "imageSrc": "user-src",
      "initials": "user-initials",
      "name": "user-name",
    }
  }
  utilityItems={
    Array [
      Object {
        "icon": <span>
          my icon
        </span>,
        "key": "my-test-key-2",
        "text": "test-text-2",
      },
    ]
  }
>
  <ApplicationLoadingOverlayProvider>
    <ApplicationStatusOverlayProvider>
      <withPromptRegistration(NavigationPromptCheckpoint)>
        <injectIntl(ApplicationErrorBoundary)>
          <Suspense
            fallback={
              <ApplicationLoadingOverlay
                backgroundStyle="clear"
                isOpen={true}
              />
            }
          >
            <div>
              test child content
            </div>
          </Suspense>
        </injectIntl(ApplicationErrorBoundary)>
      </withPromptRegistration(NavigationPromptCheckpoint)>
    </ApplicationStatusOverlayProvider>
  </ApplicationLoadingOverlayProvider>
</ApplicationNavigation>
`;
